src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +104,125 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
+
+                    LogMsalSuccessTelemetryToOTel(authenticationResult);
+
                     return authenticationResult;
                 }
                 catch (MsalException ex)
                 {
                     apiEvent.ApiErrorCode = ex.ErrorCode;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+
+                        ServiceBundle.PlatformProxy.OtelInstrumentation.LogFailedMetrics(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        ex.ErrorCode);
+                    LogMsalFailedTelemetryToOTel(ex);
                     throw;
                 }
                 catch (Exception ex)
                 {
                     apiEvent.ApiErrorCode = ex.GetType().Name;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+                    ServiceBundle.PlatformProxy.OtelInstrumentation.LogFailedMetrics(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        ex.GetType().Name);
+                    LogMsalFailedTelemetryToOTel(ex);
                     throw;
                 }
                 finally
                 {
+                    ServiceBundle.PlatformProxy.OtelInstrumentation?.StopActivity();
                     telemetryClients.TrackEvent(telemetryEventDetails);
                 }
             }
         }
 
+        private void LogMsalSuccessTelemetryToOTel(AuthenticationResult authenticationResult)
+        {
+            if (ServiceBundle.PlatformProxy.OtelInstrumentation == null)
+            {
+                return;
+            }
+
+            ServiceBundle.PlatformProxy.OtelInstrumentation.LogActivityStatus(true);
+
+            Dictionary<string, object> tags = new Dictionary<string, object> { 
+                { TelemetryConstants.CacheInfoTelemetry, Convert.ToInt64(authenticationResult.AuthenticationResultMetadata.CacheRefreshReason) },
+                { TelemetryConstants.TokenSource, Convert.ToInt64(authenticationResult.AuthenticationResultMetadata.TokenSource) },
+                { TelemetryConstants.Duration, authenticationResult.AuthenticationResultMetadata.DurationTotalInMs },
+                { TelemetryConstants.DurationInCache, authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs },
+                { TelemetryConstants.DurationInHttp, authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs },
+                { TelemetryConstants.TokenType, (int)AuthenticationRequestParameters.RequestContext.ApiEvent.TokenType },
+                { TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds } 
```

Comment:
Since this is a new telemetry store, can we improve on the key name constants and use more specific ones?
For example:
`CacheInfoTelemetry` > `CacheRefreshReason`
`Duration` > `RequestDurationInMs`
`DurationInCache` > `DurationInCacheInMs`
`DurationInHttp` > `DurationInHttpInMs`
`RemainingLifetime` > `RemainingTokenLifetimeInMs`

Reply:
Discussed offline